ext/CUDASupportExt.jl

```diff
@@ -0,0 +1,26 @@
+module CUDASupportExt
+using CUDA 
+using Adapt
+using ShiftedArrays
+using Base # to allow displaying such arrays without causing the single indexing CUDA error
+
+Adapt.adapt_structure(to, x::CircShiftedArray{T, D, CT}) where {T,D,CT<:CuArray} = CircShiftedArray(adapt(to, parent(x)), shifts(x));
```

Comment:
```suggestion
Adapt.adapt_structure(to, x::CircShiftedArray{T, D}) where {T, D} = CircShiftedArray(adapt(to, parent(x)), shifts(x));
```

I was wondering if here and below we could do this, and have it just in a Adapt extention (that way other gpu arrays types are also supported out of the box).